File: Chapter-10/kadane_algorithm.cpp
#include<iostream>
using namespace std;
int Kadanes_Algorithm(int arr[],int num) {
   int Highest_Max = 0;
   int current_MaxElement = 0;
   for(int i = 0; i < num; i++){
      current_MaxElement =max(arr[i],current_MaxElement + arr[i]) ;
      Highest_Max = max(Highest_Max,current_MaxElement);
   }
   return Highest_Max;
}
int main() {
   cout << "Enter the number of elements in the array : ";
   int num;
   cin >> num;
   int arr[num];
   cout << "Enter the elements of an array : ";
   for (int i = 0; i < num; i++) {
      cin >> arr[i];
   }
   cout << "The maximum sum is : "<<Kadanes_Algorithm(arr,num) << endl;
   return 0;
}

File: Chapter-10/kadane_algorithm.py
from sys import maxsize

def max_SubArraySum(arr, size):
    max_so_far = -maxsize - 1
    max_ending_here = 0

    for i in range(0, size):
        max_ending_here = max_ending_here + arr[i]
        if (max_so_far < max_ending_here):
            max_so_far = max_ending_here

        if max_ending_here < 0:
            max_ending_here = 0
    return max_so_far

arr = [-1,7,4,3,-2]

print("Maximum contiguous sum is : ", max_SubArraySum(arr, len(arr)))


